packages/go/analysis/ad/queries.go

```diff
@@ -1471,34 +1472,91 @@ func FetchCertTemplatesPublishedToCA(tx graph.Transaction, ca *graph.Node) (grap
 	}))
 }
 
-func FetchCanAbuseWeakCertBindingRels(tx graph.Transaction, node *graph.Node) ([]*graph.Relationship, error) {
-	if rels, err := ops.FetchRelationships(tx.Relationships().Filterf(func() graph.Criteria {
-		return query.And(
-			query.Equals(query.StartID(), node.ID),
-			query.Kind(query.Relationship(), ad.CanAbuseWeakCertBinding),
-			query.Kind(query.End(), ad.Entity),
-		)
-	})); err != nil {
-		return nil, err
+func HasUPNCertMappingInForest(tx graph.Transaction, domain *graph.Node) (bool, error) {
+	if trustedByNodes, err := FetchNodesWithTrustedByParentChildRelationship(tx, domain); err != nil {
+		log.Errorf("error in HasUPNCertMappingInForest: unable to fetch TrustedBy nodes: %v", err)
+		return false, err
 	} else {
-		return rels, nil
+		for _, trustedByDomain := range trustedByNodes {
+			if dcForNodes, err := FetchNodesWithDCForEdge(tx, trustedByDomain); err != nil {
+				log.Errorf("error in HasUPNCertMappingInForest: unable to fetch DCFor nodes: %v", err)
+				continue
+			} else {
+				for _, dcForNode := range dcForNodes {
+					if cmmrProperty, err := dcForNode.Properties.Get(ad.CertificateMappingMethodsRaw.String()).Int(); err != nil {
+						// We do not want to throw an error here as this property only exists if privileged collection has been performed
+						continue
+					} else if cmmrProperty == ein.RegistryValueDoesNotExist {
+						continue
+					} else if cmmrProperty&int(ein.CertificateMappingUserPrincipalName) == int(ein.CertificateMappingUserPrincipalName) {
+						return true, nil
+					}
+				}
+			}
+		}
 	}
+	return false, nil
 }
 
-func FetchCanAbuseUPNCertMappingRels(tx graph.Transaction, node *graph.Node) ([]*graph.Relationship, error) {
-	if rels, err := ops.FetchRelationships(tx.Relationships().Filterf(func() graph.Criteria {
-		return query.And(
-			query.Equals(query.StartID(), node.ID),
-			query.Kind(query.Relationship(), ad.CanAbuseUPNCertMapping),
-			query.Kind(query.End(), ad.Entity),
-		)
-	})); err != nil {
+func HasWeakCertBindingInForest(tx graph.Transaction, domain *graph.Node) (bool, error) {
+	if trustedByNodes, err := FetchNodesWithTrustedByParentChildRelationship(tx, domain); err != nil {
+		log.Errorf("error in HasWeakCertBindingInForest: unable to fetch TrustedBy nodes: %v", err)
```

Comment:
Since the err is returned and also logged upstream, this log seems redundant.

Reply:
I see. will fix 👍